### PR TITLE
Composer installable

### PR DESCRIPTION
**Pull request by ModulesUnraveled**
This PR does the following:
- Makes Emulsify composer installable
- Gets all of the twig functions on `composer install`
- Puts the twig functions in place on `yarn install` (or `npm install`)

## To Review:

**If installing Emulsify standalone:**
- [x] Clone emulsify and then run `composer install` inside it, or just run `composer create-project "fourkitchens/emulsify:dev-composer-installable" --stability dev --no-interaction emulsify`
- [x] Run `yarn upgrade && yarn install && yarn start`
- [x] Verify the bem and add_attributes functions are copied to `components/_twig-components/functions`
- [x] Visit the PL page and verify it loads and stuff

**If installing inside a Drupal instance:**
- [x] Install emulsify `composer require "fourkitchens/emulsify:dev-composer-installable"`
- [x] Move into the emulsify directory `cd web/themes/contrib/emulsify/`
- [x] Run `yarn upgrade && yarn install && yarn start`
- [x] Verify the bem and add_attributes functions are copied to `components/_twig-components/functions`
- [x] Visit the PL page and verify it loads and stuff

## Todo:
- [x] ~The child theme needs to get these components as well. Maybe a symlink from the child themes `functions` folder to the one in Emulsify's `vendor` directory? (Maybe emulsify should just use a symlink too, instead of copying the file.)~
  (I think as long as the package.json is copied, along with the `scripts` folder, the child theme will get the functions in place.)
- [x] ~The functions are placed in the root vendor directory, not `emulsify/vendor` when installed in a Drupal instance... What do we do?~
  (I created a script to check for the vendor directories, so it'll work in both cases)
- [ ] Drush command needs to copy the `scripts` directory for both regular and --slim

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -2,39 +2,43 @@
...
     "require-dev": {},
     "minimum-stability": "dev",
     "prefer-stable": true,
     "scripts": {
         "pre-install-cmd": [],
         "pre-update-cmd": [],
-        "post-install-cmd": [],
-        "post-update-cmd": []
+        "post-install-cmd": [
+          "drush emulsify"
```
Comment (ccjjmartin) [inline composer.json]: Remove

File: composer.json [hunk 2]
```diff
@@ -2,39 +2,43 @@
...
-            "role": ""
+            "email": "randy@fourkitchens.com"
         }
     ],
-    "repositories": [],
-    "require": {},
+    "repositories": {},
+    "require": {
+        "drupal/components": "^1.0",
+        "drupal-pattern-lab/unified-twig-extensions": "dev-develop"
```
Comment (ccjjmartin) [inline composer.json]: Can this be on drupal.org?

Comment (ccjjmartin) [inline composer.json]: Sounds like it will be on Packagist

File: scripts/twig_functions.sh [hunk 3]
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+# Link contrib twig functions into the components directory
+
+# First we need to find out whether this is inside a Drupal install or standalone Emulsify
+if [ -d ../../../../vendor ]
```
Comment (evanmwillhite) [inline scripts/twig_functions.sh]: I have an older Drupal install that I ran this inside of and it didn't have the /web directory. Should we make this more open-ended to handle that?

File: README.md [hunk 4]
```diff
@@ -11,19 +11,54 @@ Component-driven prototyping tool using [Pattern Lab v2](http://patternlab.io/)
...
+  3. `npm install` or `yarn install`
+
+## Drupal-specific installation
+
+### In a Composer-based Drupal install (recommended)
+
+  1. `composer require fourkitchens/emulsify`
+  2. `cd web/themes/contrib/emulsify/`
+  3. `npm install` or `yarn install`
+  4. Optional (but recommeded): Create "child" theme `drush emulsify "Theme Name"` (more details below)
```
Comment (evanmwillhite) [inline README.md]: *recommended

Comment (evanmwillhite) [inline README.md]: Also, is it `drush emulsify "Theme Name"` or `drush emulsify site_machine_name`

**Pull request merged**